### ISSUE TITLE
FYI: recent optimization in core JAX rfft transpose implementation

### ISSUE THREAD

**Issue by michael-0brien** [flatironinstitute/jax-finufft#203]
Hi, wanted to pass this along in case devs haven't seen: recently a patch was made to the core JAX rfftn transpose implementation: https://github.com/jax-ml/jax/discussions/34553. This meant that transpose calls to `jax.numpy.fft.rfftn` were always using a full grid, rather than taking advantage of hermitian symmetry.

Bringing this to people's attention just to alert that libraries using `jax-finufft` (such as mine) will want to soon upgrade to the most recent version of JAX!
